iterators.tex

```diff
@@ -6394,26 +6397,32 @@
 only if \tcode{size(t)} returns the number of elements in \tcode{t}.}
 
 \pnum
-For any type \tcode{T}, \tcode{is_sized_range<T>} derives from
-\tcode{true_type} if \tcode{T} \oldtxt{models}\newtxt{meets the syntactic requirements of}
-\tcode{\xname{SizedRangeLike}}, and \tcode{false_type} otherwise.
+For a\oldtxt{ny} type \tcode{T} \newtxt{that satisfies \tcode{\xname{SizedRangeLike<T>}}},
+\tcode{is_sized_range<T>} derives from \tcode{true_type} if \oldtxt{\tcode{T} models
+\tcode{\xname{SizedRangeLike}}} \newtxt{\tcode{Same<T, remove\_cv\_t<remove\_reference\_t<T>{}>{}>()} is satisfied}, and \oldtxt{\tcode{false_type}} \newtxt{\tcode{is_sized_range<remove_cv_t<remove_reference_t<T>{}>{}>}} otherwise.
```

Comment:
I suppose the alternative would be to specify the requirements directly in the text instead of naming `SizedRangeLike`, but eww no.

It's part of the design whether the name starts with _ or not. It seems odd to me to fully specify this concept which users critically must understand to avoid undefined behavior and every implementation needs to implement almost exactly as specified yet keep it exposition only. It's there, I know that it exists and I have to conform to it, but I can't rely on the name.

Of course we wouldn't need it at all if we changed to a pure opt-out trait:
```
template <class R>
struct disable_sized_range : std::false_type {};
template <class R>
concept bool SizedRange() {
  return _IsNot<__uncvref<R>, disable_sized_range> && Range<R> && requires (const R& r) {
    { size(r) } -> Integral;
    { size(r) } -> ConvertibleTo<DifferenceType<IteratorType<R&>>>;
  };
}
```
Gosh that sure does make the design simple ;)